### PR TITLE
Docs/add documentation

### PR DESCRIPTION
**Pull request by tm-kyle-naranjo**
## Type of PR
Documentation

## Description
Add documentation for the following:
- architecture
- development
- troubleshooting

## How to check
1. Go to docs/
2. Review the architecture.md/development.md/troubleshooting.md

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -1,49 +1,117 @@
...
+
+## References
+
+* [Dagster documentation](https://docs.dagster.io/)
+
+## CI/CD Pipeline
+
+The CI/CD pipeline will automatically build, test, and deploy the tagged version to the staging environment. After testing in staging, manually promote to production using the AWS CodePipeline console.
+
+Since CI/CD is integrated, you can simply run the CodePipeline to have the changes automatically deployed to the staging environment. Once deployed, you can access the application on the ALB domain at http://dwh-alb-jjk-963539840.ap-southeast-1.elb.amazonaws.com.
```
Comment (tm-echo-arcelao) [inline docs/development.md:117]: reminder that we will take down the resources after this lab. thus, this endpoint will be inaccessible.

File: docs/architecture.md [hunk 2]
```diff
@@ -1,101 +1,103 @@
...
+   - Click "Connect" to open a terminal session directly in your browser.
+
+This method provides secure access to the EC2 instance without needing to configure SSH or use a VPN.
+
+### Access Dagster Dagit UI
+
+**Navigate to the ALB Domain**:
+
+The Dagit UI is available on the ALB domain. To access it, navigate to the following URL:
+[http://dwh-alb-jjk-963539840.ap-southeast-1.elb.amazonaws.com](http://dwh-alb-jjk-963539840.ap-southeast-1.elb.amazonaws.com)
```
Comment (tm-echo-arcelao) [inline docs/architecture.md:70]: assume that this endpoint is no longer accessible after the lab.

Comment (tm-kyle-naranjo) [reply docs/architecture.md]: Resolved in this comment [c3fca59](https://github.com/tm-echo-arcelao/te-jjk-platforms-cloud-lab/pull/6/commits/c3fca593be1c0dee862d229367456e2c2a94a2ef)

File: docs/development.md (hunk 1 above)
Comment (tm-kyle-naranjo) [reply docs/development.md]: Resolved in this commit [c3fca59](https://github.com/tm-echo-arcelao/te-jjk-platforms-cloud-lab/pull/6/commits/c3fca593be1c0dee862d229367456e2c2a94a2ef)

**Review by tm-echo-arcelao — APPROVED** [review]

**Comment by tm-echo-arcelao** [comment]
@tm-kyle-naranjo merge when ready 😄

**Pull request merged**